src/polygon/src/polygon.cpp

```diff
@@ -344,7 +345,10 @@ class Monotones {
     bool IsPast(const VertItr other, float precision) const {
       return pos.y > other->pos.y + precision;
     }
-    bool operator<(const VertAdj &other) const { return pos.y < other.pos.y; }
+    bool operator<(const VertAdj &other) const {
+      return pos.y < other.pos.y ||
+             (pos.y == other.pos.y && insertionID < other.insertionID);
```

Comment:
So why is it important to make the priority queue stable? Is this about correctness or just determinism? Why was the new polygon test failing in the first place?

Reply:
It can at least improve determinism. I am not sure if it is more correct as I haven't yet look into the details of the triangulation algorithm. I was comparing the run of wasm and native and found that the wasm one got unstable priority queue, and changing that fixes this particular test.

Reply:
I did some other tests, it seems that the stability of the priority queue is not the fundamental issue, I guess I should try to understand the triangulation algorithm and see how make some minimal example.

Reply:
That makes sense. Yeah, my triangulator is a bit convoluted - believe it or not I've refactored it multiple times to try to simplify it. Would be great to have another set of eyes on it. Let me know if you have any questions.